p2p/mock/peer.go

```diff
@@ -40,6 +40,24 @@ func NewPeer(ip net.IP) *Peer {
 	return mp
 }
 
+func NewFixIdPeer(ip net.IP, id p2p.ID) *Peer {
```

Comment:
Can't we use `NewPeer`? Why need to provide an external ID?

Reply:
Can't we overwrite ID in any case?

```go
peer = mock.NewPeer(nil)
peer.ID = nodeID
```

Reply:
It is an option, but we should expose ID.

Reply:
lets do that. it's a mock peer anyway